test/dst/dst.go

```diff
@@ -126,11 +127,10 @@ func (d *DST) Run(r *rand.Rand, api api.API, aio aio.AIO, system *system.System,
 				Metadata:   metadata,
 				Submission: req,
 				Callback: func(res *t_api.Response, err error) {
-					modelErr := model.Step(t, req, res, err)
+					modelErr := model.Step(t, req, res, err, d.config.FaultInjectionMode)
```

Comment:
Rather than pass this in for every step, let's instantiate the model with the "scenario" information since this does not change. I think we could do that [here](https://github.com/resonatehq/resonate/blob/main/test/dst/dst.go#L45).